third_party/terraform/data_sources/data_source_google_iam_testable_permissions.go

```diff
@@ -0,0 +1,157 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceGoogleIamTestablePermissions() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleIamTestablePermissionsRead,
+		Schema: map[string]*schema.Schema{
+			"full_resource_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"custom_support_level": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "SUPPORTED",
+			},
+			"stage": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "GA",
+			},
+			"permissions": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"title": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"custom_support_level": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"stage": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"api_disabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceGoogleIamTestablePermissionsRead(d *schema.ResourceData, meta interface{}) (err error) {
+	config := meta.(*Config)
+	body := make(map[string]interface{})
+	body["pageSize"] = 500
+	permissions := make([]map[string]interface{}, 0)
+	custom_support_level := d.Get("custom_support_level").(string)
+	if err = validatePermissionCustomSupport(custom_support_level); err != nil {
```

Comment:
You can add this type of validation on the schema directly. See https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_container_cluster.go#L388

Reply:
Awesome! Will update